src/test/java/duke/parser/CommandParserTest.java

```diff
@@ -0,0 +1,66 @@
+package duke.parser;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import org.junit.jupiter.api.Test;
+
+import duke.commands.Command;
+
+public class CommandParserTest {
+
+    @Test
+    public void todoParseTest() {
+        CommandParser cp = new CommandParser();
+        String testInput = "todo test123 test456";
+        Command cmd = cp.parseCommand(testInput);
+        assertNotNull(cmd);
+        assertEquals(testInput, cmd.toString());
+    }
+
+    @Test
+    public void deadlineParseTest() {
+        CommandParser cp = new CommandParser();
+        String testInput = "deadline test123 test456 /by 3/8/2023 1200";
+        Command cmd = cp.parseCommand(testInput);
+        assertNotNull(cmd);
+        assertEquals(testInput, cmd.toString());
+    }
+
+    @Test
+    public void eventParseTest() {
+        CommandParser cp = new CommandParser();
+        String testInput = "event test123 test456 /from 3/8/2023 1200 /to 5/8/2023 1200";
+        Command cmd = cp.parseCommand(testInput);
+        assertNotNull(cmd);
+        assertEquals(testInput, cmd.toString());
+    }
+
+    @Test
+    public void invalidParseTest() {
+        CommandParser cp = new CommandParser();
+        String testInput = "test123 test456";
+        Command cmd = cp.parseCommand(testInput);
+        assertNotNull(cmd);
+        assertEquals("Invalid Command! Command not found", cmd.toString());
+    }
+
+    @Test
+    public void invalidParseTest2() {
```

Comment:
Perhaps it might be good to leave a bit of comment for the invalidParseTest2 to show the difference with invalidParseTest.